db/migrate/20151008034310_add_index_to_semester.rb

```diff
@@ -0,0 +1,5 @@
+class AddIndexToSemester < ActiveRecord::Migration
+  def change
+    add_index :semesters, [:year, :season], unique: true
```

Comment:
I think in the big picture we  only need to index year since there are 4 seasons per year tops (unless I misunderstand this) so the year index is enough.  Do you plan on season level aggregation?